packages/block-editor/src/components/global-styles/typography-panel.js

```diff
@@ -107,27 +80,16 @@ function useAppearanceControlLabel( name, element, settings ) {
 	return __( 'Appearance' );
 }
 
-function useHasLetterSpacingControl( name, element, settings ) {
-	const setting = settings?.typography?.letterSpacing;
-	const supports = useSupportedStyles( name, element );
-	if ( ! setting ) {
-		return false;
-	}
-	return supports.includes( 'letterSpacing' );
+function useHasLetterSpacingControl( settings ) {
+	return settings?.typography?.letterSpacing;
 }
 
-function useHasTextTransformControl( name, element, settings ) {
-	const setting = settings?.typography?.textTransform;
-	const supports = useSupportedStyles( name, element );
-	if ( ! setting ) {
-		return false;
-	}
-	return supports.includes( 'textTransform' );
+function useHasTextTransformControl( settings ) {
+	return settings?.typography?.textTransform;
 }
 
-function useHasTextDecorationControl( name, element ) {
-	const supports = useSupportedStyles( name, element );
-	return supports.includes( 'textDecoration' );
```

Comment:
Notice that the previous implementation of text transform didn't take into consideration the "setting" value that comes from theme.json. I think this is an inconsistency and probably something that was "forgotten". So in that sense, the current PR changes the behavior a bit but I believe it's better anyway.